ipaserver/plugins/server.py

```diff
@@ -659,10 +659,24 @@ def _remove_server_host_services(self, ldap, master):
         delete server kerberos key and all its svc principals
         """
         try:
-            entries = ldap.get_entries(
-                self.api.env.basedn, ldap.SCOPE_SUBTREE,
+            # do not delete ldap principal if server-del command 
+            # has been called on a machine which is being deleted
+            # since this will break replication.
+            # ldap principal to be cleaned later by topology plugin
+            if master == self. api.env.host:
+                filter=(
```

Comment:
In this filter, it is better to swap internal filters so that we first reduce the search space to the set of principals from the same host and then eliminate `ldap/..` principal from the list.

Reply:
done